
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>EnsemblRest &#8212; EnsemblRest 0.1.3 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1.3',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="ensemblrest">
<h1>EnsemblRest<a class="headerlink" href="#ensemblrest" title="Permalink to this headline">¶</a></h1>
<a class="reference external image-reference" href="https://saythanks.io/to/Ad115"><img alt="https://img.shields.io/badge/Say%20Thanks-!-1EAEDB.svg" src="https://img.shields.io/badge/Say%20Thanks-!-1EAEDB.svg" /></a>
<p>A Python interface to the Ensembl REST APIs. A whole world of biological data
at your fingertips.</p>
<p>The <a class="reference external" href="https://www.ensembl.org/index.html">Ensembl database</a> contains
reference biological data on almost any organism. Now it is easy to
access this data programatically through their REST API.</p>
<p>The specific documentation on the Ensembl REST API endpoints is on <a class="reference external" href="https://rest.ensembl.org/">their
website</a>.</p>
<p>This library also includes some utilities built on top of the APIs designed to
ease working with them, including an
<a class="reference external" href="https://ad115.github.io/EnsemblRest/#ensembl_rest.AssemblyMapper">AssemblyMapper</a>
class that helps in the conversion between different genome assemblies.</p>
<p>This project uses code from <a class="reference external" href="https://github.com/rapidstack/RESTEasy">RESTEasy</a>,
which made my life much easier. Thanks!</p>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<p>You can install from <a class="reference external" href="https://pypi.org/project/ensembl-rest/">PyPI</a>:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>$ pip install ensembl_rest
</pre></div>
</div>
</div>
</div>
<div class="section" id="examples">
<h1>Examples<a class="headerlink" href="#examples" title="Permalink to this headline">¶</a></h1>
<p>The library exports two main classes: <code class="docutils literal"><span class="pre">ensembl_rest.EnsemblClient</span></code> and
<code class="docutils literal"><span class="pre">ensembl_rest.EnsemblGenomesClient</span></code> that point respectively to the <a class="reference external" href="http://rest.ensembl.org/">main
REST API</a> and to the <a class="reference external" href="http://rest.ensemblgenomes.org/">Ensembl Genomes REST
API</a>.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">import</span> <span class="nn">ensembl_rest</span>

<span class="gp">&gt;&gt;&gt; </span><span class="n">client</span> <span class="o">=</span> <span class="n">ensembl_rest</span><span class="o">.</span><span class="n">EnsemblClient</span><span class="p">()</span>
</pre></div>
</div>
<p>If you want to use a method from the REST API, say:
<code class="docutils literal"><span class="pre">GET</span> <span class="pre">lookup/symbol/:species/:symbol</span></code>
(<a class="reference external" href="http://rest.ensembl.org/documentation/info/symbol_lookup">http://rest.ensembl.org/documentation/info/symbol_lookup</a>)
then the corresponding method on the class is called after the last
string in the link to the documentation page, in this case, <code class="docutils literal"><span class="pre">symbol_lookup</span></code>.</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">help</span><span class="p">(</span><span class="n">client</span><span class="o">.</span><span class="n">symbol_lookup</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-default"><div class="highlight"><pre><span></span>Help on method symbol_lookup in module ensembl_rest.core.baseclient:

symbol_lookup(*args, **kwargs) method of ensembl_rest.core.baseclient.EnsemblClient instance
    ``GET lookup/symbol/:species/:symbol``

    Find the species and database for a symbol in a linked external database
    - More info: https://rest.ensembl.org/documentation/info/symbol_lookup
</pre></div>
</div>
<p>We can see from the resource string
<code class="docutils literal"><span class="pre">GET</span> <span class="pre">lookup/symbol/:species/:symbol</span></code> that this method contains 2
parameters called species and symbol, so we can call the method in the
following way:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">client</span><span class="o">.</span><span class="n">symbol_lookup</span><span class="p">(</span><span class="n">species</span><span class="o">=</span><span class="s1">&#39;homo sapiens&#39;</span><span class="p">,</span>
<span class="go">                         symbol=&#39;TP53&#39;)</span>

<span class="go"># Or like this...</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">client</span><span class="o">.</span><span class="n">symbol_lookup</span><span class="p">(</span><span class="s1">&#39;homo sapiens&#39;</span><span class="p">,</span> <span class="s1">&#39;TP53&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="p">{</span><span class="s1">&#39;source&#39;</span><span class="p">:</span> <span class="s1">&#39;ensembl_havana&#39;</span><span class="p">,</span>
 <span class="s1">&#39;object_type&#39;</span><span class="p">:</span> <span class="s1">&#39;Gene&#39;</span><span class="p">,</span>
 <span class="s1">&#39;logic_name&#39;</span><span class="p">:</span> <span class="s1">&#39;ensembl_havana_gene&#39;</span><span class="p">,</span>
 <span class="s1">&#39;version&#39;</span><span class="p">:</span> <span class="mi">16</span><span class="p">,</span>
 <span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;homo sapiens&#39;</span><span class="p">,</span>
 <span class="s1">&#39;description&#39;</span><span class="p">:</span> <span class="s1">&#39;tumor protein p53 [Source:HGNC Symbol;Acc:HGNC:11998]&#39;</span><span class="p">,</span>
 <span class="s1">&#39;display_name&#39;</span><span class="p">:</span> <span class="s1">&#39;TP53&#39;</span><span class="p">,</span>
 <span class="s1">&#39;assembly_name&#39;</span><span class="p">:</span> <span class="s1">&#39;GRCh38&#39;</span><span class="p">,</span>
 <span class="s1">&#39;biotype&#39;</span><span class="p">:</span> <span class="s1">&#39;protein_coding&#39;</span><span class="p">,</span>
 <span class="s1">&#39;end&#39;</span><span class="p">:</span> <span class="mi">7687550</span><span class="p">,</span>
 <span class="s1">&#39;seq_region_name&#39;</span><span class="p">:</span> <span class="s1">&#39;17&#39;</span><span class="p">,</span>
 <span class="s1">&#39;db_type&#39;</span><span class="p">:</span> <span class="s1">&#39;core&#39;</span><span class="p">,</span>
 <span class="s1">&#39;strand&#39;</span><span class="p">:</span> <span class="o">-</span><span class="mi">1</span><span class="p">,</span>
 <span class="s1">&#39;id&#39;</span><span class="p">:</span> <span class="s1">&#39;ENSG00000141510&#39;</span><span class="p">,</span>
 <span class="s1">&#39;start&#39;</span><span class="p">:</span> <span class="mi">7661779</span><span class="p">}</span>
</pre></div>
</div>
<p>Another common usage is to fetch sequences of known genes:</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">client</span><span class="o">.</span><span class="n">sequence_id</span><span class="p">(</span><span class="s1">&#39;ENSG00000157764&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="p">{</span><span class="s1">&#39;desc&#39;</span><span class="p">:</span> <span class="s1">&#39;chromosome:GRCh38:7:140719327:140924928:-1&#39;</span><span class="p">,</span>
 <span class="s1">&#39;query&#39;</span><span class="p">:</span> <span class="s1">&#39;ENSG00000157764&#39;</span><span class="p">,</span>
 <span class="s1">&#39;version&#39;</span><span class="p">:</span> <span class="mi">13</span><span class="p">,</span>
 <span class="s1">&#39;id&#39;</span><span class="p">:</span> <span class="s1">&#39;ENSG00000157764&#39;</span><span class="p">,</span>
 <span class="s1">&#39;seq&#39;</span><span class="p">:</span> <span class="s1">&#39;TTCCCCCAATCCCCTCAGGCTCGGCTGCGCCCGGGGC...ACTGCTATAATAAAGATTGACTGCATGGAGAAGTCTTCA&#39;</span><span class="p">,</span>
 <span class="s1">&#39;molecule&#39;</span><span class="p">:</span> <span class="s1">&#39;dna&#39;</span><span class="p">}</span>
</pre></div>
</div>
<p>Or to map betweeen assemblies…</p>
<div class="highlight-python"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">client</span><span class="o">.</span><span class="n">assembly_map</span><span class="p">(</span><span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span>
<span class="go">                        asm_one=&#39;GRCh37&#39;,</span>
<span class="go">                        region=&#39;X:1000000..1000100:1&#39;,</span>
<span class="go">                        asm_two=&#39;GRCh38&#39;)</span>


<span class="go"># Or...</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">region_str</span> <span class="o">=</span> <span class="n">ensembl_rest</span><span class="o">.</span><span class="n">region_str</span><span class="p">(</span><span class="n">chom</span><span class="o">=</span><span class="s1">&#39;X&#39;</span><span class="p">,</span>
<span class="go">                                         start=1000000,</span>
<span class="go">                                         end=1000100)</span>

<span class="gp">&gt;&gt;&gt; </span><span class="n">client</span><span class="o">.</span><span class="n">assembly_map</span><span class="p">(</span><span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span>
<span class="go">                        asm_one=&#39;GRCh37&#39;,</span>
<span class="go">                        region=region_str,</span>
<span class="go">                        asm_two=&#39;GRCh38&#39;)</span>
</pre></div>
</div>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="p">{</span><span class="s1">&#39;mappings&#39;</span><span class="p">:</span> <span class="p">[{</span><span class="s1">&#39;original&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;seq_region_name&#39;</span><span class="p">:</span> <span class="s1">&#39;X&#39;</span><span class="p">,</span>
    <span class="s1">&#39;strand&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="s1">&#39;coord_system&#39;</span><span class="p">:</span> <span class="s1">&#39;chromosome&#39;</span><span class="p">,</span>
    <span class="s1">&#39;end&#39;</span><span class="p">:</span> <span class="mi">1000100</span><span class="p">,</span>
    <span class="s1">&#39;start&#39;</span><span class="p">:</span> <span class="mi">1000000</span><span class="p">,</span>
    <span class="s1">&#39;assembly&#39;</span><span class="p">:</span> <span class="s1">&#39;GRCh37&#39;</span><span class="p">},</span>
   <span class="s1">&#39;mapped&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;seq_region_name&#39;</span><span class="p">:</span> <span class="s1">&#39;X&#39;</span><span class="p">,</span>
    <span class="s1">&#39;strand&#39;</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span>
    <span class="s1">&#39;coord_system&#39;</span><span class="p">:</span> <span class="s1">&#39;chromosome&#39;</span><span class="p">,</span>
    <span class="s1">&#39;end&#39;</span><span class="p">:</span> <span class="mi">1039365</span><span class="p">,</span>
    <span class="s1">&#39;start&#39;</span><span class="p">:</span> <span class="mi">1039265</span><span class="p">,</span>
    <span class="s1">&#39;assembly&#39;</span><span class="p">:</span> <span class="s1">&#39;GRCh38&#39;</span><span class="p">}}]}</span>
</pre></div>
</div>
<p>The above problem (mapping from one assembly to another) is so frequent that
the library provides a specialized class <code class="docutils literal"><span class="pre">AssemblyMapper</span></code> to efficiently
mapping large amounts of regions between assemblies. This class avoids the
time-consuming task of making a web request every time a mapping is needed by
fetching the mapping of the whole assembly right from the instantiation. This
is a time-consuming operation by itself, but it pays off when one has to
transform repeatedly betweeen assemblies.:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">mapper</span> <span class="o">=</span> <span class="n">ensembl_rest</span><span class="o">.</span><span class="n">AssemblyMapper</span><span class="p">(</span><span class="n">from_assembly</span><span class="o">=</span><span class="s1">&#39;GRCh37&#39;</span>
<span class="gp">... </span>                                     <span class="n">to_assembly</span><span class="o">=</span><span class="s1">&#39;GRCh38&#39;</span><span class="p">)</span>

<span class="gp">&gt;&gt;&gt; </span><span class="n">mapper</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="n">chrom</span><span class="o">=</span><span class="s1">&#39;1&#39;</span><span class="p">,</span> <span class="n">pos</span><span class="o">=</span><span class="mi">1000000</span><span class="p">)</span>
<span class="go">1064620</span>
</pre></div>
</div>
</div>
<div class="section" id="meta">
<h1>Meta<a class="headerlink" href="#meta" title="Permalink to this headline">¶</a></h1>
<p><strong>Author</strong>: <a class="reference external" href="https://agargar.wordpress.com/">Ad115</a> -
<a class="reference external" href="https://github.com/Ad115/">Github</a> – <a class="reference external" href="mailto:a&#46;garcia230395&#37;&#52;&#48;gmail&#46;com">a<span>&#46;</span>garcia230395<span>&#64;</span>gmail<span>&#46;</span>com</a></p>
<p>Distributed under the MIT license. See
<a class="reference external" href="https://github.com/Ad115/PyEnsembl/blob/master/LICENSE">LICENSE</a>
for more information.</p>
</div>
<div class="section" id="contributing">
<h1>Contributing<a class="headerlink" href="#contributing" title="Permalink to this headline">¶</a></h1>
<ol class="arabic simple">
<li>Check for open issues or open a fresh issue to start a discussion
around a feature idea or a bug.</li>
<li>Fork <a class="reference external" href="https://github.com/Ad115/EnsemblRest/">the repository</a>
on GitHub to start making your changes to a feature branch, derived
from the <strong>master</strong> branch.</li>
<li>Write a test which shows that the bug was fixed or that the feature
works as expected.</li>
<li>Send a pull request and bug the maintainer until it gets merged and
published.</li>
</ol>
</div>
<hr class="docutils" />
<div class="section" id="api-documentation">
<h1>API Documentation<a class="headerlink" href="#api-documentation" title="Permalink to this headline">¶</a></h1>
<dl class="function">
<dt id="ensembl_rest.region_str">
<code class="descclassname">ensembl_rest.</code><code class="descname">region_str</code><span class="sig-paren">(</span><em>chrom</em>, <em>start</em>, <em>end=None</em>, <em>strand=1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ensembl_rest/utils.html#region_str"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ensembl_rest.region_str" title="Permalink to this definition">¶</a></dt>
<dd><p>Assemble a region string suitable for consumption for the Ensembl REST API.</p>
<p>The generated string has the format: <code class="docutils literal"><span class="pre">{chrom}:{start}..{end}:{strand}</span></code></p>
</dd></dl>

<dl class="class">
<dt id="ensembl_rest.AssemblyMapper">
<em class="property">class </em><code class="descclassname">ensembl_rest.</code><code class="descname">AssemblyMapper</code><span class="sig-paren">(</span><em>from_assembly='GRCh37'</em>, <em>to_assembly='GRCh38'</em>, <em>species='human'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ensembl_rest/assemblymapper.html#AssemblyMapper"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ensembl_rest.AssemblyMapper" title="Permalink to this definition">¶</a></dt>
<dd><p>Structure that optimizes the mapping between 
diferent genome assemblies.</p>
<p>Example:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">mapper</span> <span class="o">=</span> <span class="n">AssemblyMapper</span><span class="p">(</span><span class="n">from_assembly</span><span class="o">=</span><span class="s1">&#39;GRCh37&#39;</span>
<span class="gp">... </span>                        <span class="n">to_assembly</span><span class="o">=</span><span class="s1">&#39;GRCh38&#39;</span><span class="p">)</span>

<span class="gp">&gt;&gt;&gt; </span><span class="n">mapper</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="n">chrom</span><span class="o">=</span><span class="s1">&#39;1&#39;</span><span class="p">,</span> <span class="n">pos</span><span class="o">=</span><span class="mi">1000000</span><span class="p">)</span>
<span class="go">1064620</span>
</pre></div>
</div>
<dl class="method">
<dt id="ensembl_rest.AssemblyMapper.map">
<code class="descname">map</code><span class="sig-paren">(</span><em>chrom</em>, <em>pos</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/ensembl_rest/assemblymapper.html#AssemblyMapper.map"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#ensembl_rest.AssemblyMapper.map" title="Permalink to this definition">¶</a></dt>
<dd><p>Map the given position.</p>
<p>The mapping is between the specified assemblies
when creating the object. (default: map position 
from assembly GRCh37 to assembly GRCh38)</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ensembl_rest.EnsemblClient">
<em class="property">class </em><code class="descclassname">ensembl_rest.</code><code class="descname">EnsemblClient</code><span class="sig-paren">(</span><em>base_url=None</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient" title="Permalink to this definition">¶</a></dt>
<dd><p>A client for the Ensembl REST API (<a class="reference external" href="https://rest.ensembl.org/">https://rest.ensembl.org/</a>)</p>
<dl class="method">
<dt id="ensembl_rest.EnsemblClient.VariantAnnotationSet">
<code class="descname">VariantAnnotationSet</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.VariantAnnotationSet" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/variantannotationsets/search</span></code></p>
<p>Return a list of annotation sets in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/VariantAnnotationSet">https://rest.ensembl.org/documentation/info/VariantAnnotationSet</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.VariantAnnotationSet_id">
<code class="descname">VariantAnnotationSet_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.VariantAnnotationSet_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/variantannotationsets/:id</span></code></p>
<p>Return meta data for a specific annotation set in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/VariantAnnotationSet_id">https://rest.ensembl.org/documentation/info/VariantAnnotationSet_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.analysis">
<code class="descname">analysis</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.analysis" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/analysis/:species</span></code></p>
<p>List the names of analyses involved in generating Ensembl data.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/analysis">https://rest.ensembl.org/documentation/info/analysis</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.archive_id_get">
<code class="descname">archive_id_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.archive_id_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">archive/id/:id</span></code></p>
<p>Uses the given identifier to return the archived sequence
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/archive_id_get">https://rest.ensembl.org/documentation/info/archive_id_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.archive_id_post">
<code class="descname">archive_id_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.archive_id_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">archive/id</span></code></p>
<p>Retrieve the archived sequence for a set of identifiers
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/archive_id_post">https://rest.ensembl.org/documentation/info/archive_id_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.array">
<code class="descname">array</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.array" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/microarray/:microarray/vendor/:vendor</span></code></p>
<p>Returns information about a specific microarray
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/array">https://rest.ensembl.org/documentation/info/array</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.assembly_cdna">
<code class="descname">assembly_cdna</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.assembly_cdna" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">map/cdna/:id/:region</span></code></p>
<p>Convert from cDNA coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/assembly_cdna">https://rest.ensembl.org/documentation/info/assembly_cdna</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.assembly_cds">
<code class="descname">assembly_cds</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.assembly_cds" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">map/cds/:id/:region</span></code></p>
<p>Convert from CDS coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/assembly_cds">https://rest.ensembl.org/documentation/info/assembly_cds</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.assembly_info">
<code class="descname">assembly_info</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.assembly_info" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/assembly/:species</span></code></p>
<p>List the currently available assemblies for a species, along with toplevel sequences, chromosomes and cytogenetic bands.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/assembly_info">https://rest.ensembl.org/documentation/info/assembly_info</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.assembly_map">
<code class="descname">assembly_map</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.assembly_map" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">map/:species/:asm_one/:region/:asm_two</span></code></p>
<p>Convert the co-ordinates of one assembly to another
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/assembly_map">https://rest.ensembl.org/documentation/info/assembly_map</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.assembly_stats">
<code class="descname">assembly_stats</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.assembly_stats" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/assembly/:species/:region_name</span></code></p>
<p>Returns information about the specified toplevel sequence region for the given species.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/assembly_stats">https://rest.ensembl.org/documentation/info/assembly_stats</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.assembly_translation">
<code class="descname">assembly_translation</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.assembly_translation" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">map/translation/:id/:region</span></code></p>
<p>Convert from protein (translation) coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/assembly_translation">https://rest.ensembl.org/documentation/info/assembly_translation</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.beacon_get">
<code class="descname">beacon_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.beacon_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/beacon</span></code></p>
<p>Return Beacon information
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/beacon_get">https://rest.ensembl.org/documentation/info/beacon_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.beacon_query_get">
<code class="descname">beacon_query_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.beacon_query_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/beacon/query</span></code></p>
<p>Return the Beacon response for allele information
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/beacon_query_get">https://rest.ensembl.org/documentation/info/beacon_query_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.beacon_query_post">
<code class="descname">beacon_query_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.beacon_query_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/beacon/query</span></code></p>
<p>Return the Beacon response for allele information
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/beacon_query_post">https://rest.ensembl.org/documentation/info/beacon_query_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.biotypes">
<code class="descname">biotypes</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.biotypes" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/biotypes/:species</span></code></p>
<p>List the functional classifications of gene models that Ensembl associates with a particular species. Useful for restricting the type of genes/transcripts retrieved by other endpoints.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/biotypes">https://rest.ensembl.org/documentation/info/biotypes</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.biotypes_groups">
<code class="descname">biotypes_groups</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.biotypes_groups" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/biotypes/groups/:group/:object_type</span></code></p>
<p>Without argument the list of available biotype groups is returned. With :group argument provided, list the properties of biotypes within that group. Object type (gene or transcript) can be provided for filtering.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/biotypes_groups">https://rest.ensembl.org/documentation/info/biotypes_groups</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.biotypes_name">
<code class="descname">biotypes_name</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.biotypes_name" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/biotypes/name/:name/:object_type</span></code></p>
<p>List the properties of biotypes with a given name. Object type (gene or transcript) can be provided for filtering.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/biotypes_name">https://rest.ensembl.org/documentation/info/biotypes_name</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.cafe_tree">
<code class="descname">cafe_tree</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.cafe_tree" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">cafe/genetree/id/:id</span></code></p>
<p>Retrieves a cafe tree of the gene tree using the gene tree stable identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/cafe_tree">https://rest.ensembl.org/documentation/info/cafe_tree</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.cafe_tree_member_id">
<code class="descname">cafe_tree_member_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.cafe_tree_member_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">cafe/genetree/member/id/:id</span></code></p>
<p>Retrieves the cafe tree of the gene tree that contains the gene / transcript / translation stable identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/cafe_tree_member_id">https://rest.ensembl.org/documentation/info/cafe_tree_member_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.cafe_tree_member_symbol">
<code class="descname">cafe_tree_member_symbol</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.cafe_tree_member_symbol" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">cafe/genetree/member/symbol/:species/:symbol</span></code></p>
<p>Retrieves the cafe tree of the gene tree that contains the gene identified by a symbol
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/cafe_tree_member_symbol">https://rest.ensembl.org/documentation/info/cafe_tree_member_symbol</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.compara_methods">
<code class="descname">compara_methods</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.compara_methods" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/compara/methods</span></code></p>
<p>List all compara analyses available (an analysis defines the type of comparative data).
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/compara_methods">https://rest.ensembl.org/documentation/info/compara_methods</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.compara_species_sets">
<code class="descname">compara_species_sets</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.compara_species_sets" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/compara/species_sets/:method</span></code></p>
<p>List all collections of species analysed with the specified compara method.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/compara_species_sets">https://rest.ensembl.org/documentation/info/compara_species_sets</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.comparas">
<code class="descname">comparas</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.comparas" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/comparas</span></code></p>
<p>Lists all available comparative genomics databases and their data release.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/comparas">https://rest.ensembl.org/documentation/info/comparas</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.data">
<code class="descname">data</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.data" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/data</span></code></p>
<p>Shows the data releases available on this REST server. May return more than one release (unfrequent non-standard Ensembl configuration).
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/data">https://rest.ensembl.org/documentation/info/data</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.external_dbs">
<code class="descname">external_dbs</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.external_dbs" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/external_dbs/:species</span></code></p>
<p>Lists all available external sources for a species.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/external_dbs">https://rest.ensembl.org/documentation/info/external_dbs</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.family">
<code class="descname">family</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.family" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">family/id/:id</span></code></p>
<p>Retrieves a family information using the family stable identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/family">https://rest.ensembl.org/documentation/info/family</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.family_member_id">
<code class="descname">family_member_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.family_member_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">family/member/id/:id</span></code></p>
<p>Retrieves the information for all the families that contains the gene / transcript / translation stable identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/family_member_id">https://rest.ensembl.org/documentation/info/family_member_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.family_member_symbol">
<code class="descname">family_member_symbol</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.family_member_symbol" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">family/member/symbol/:species/:symbol</span></code></p>
<p>Retrieves the information for all the families that contains the gene identified by a symbol
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/family_member_symbol">https://rest.ensembl.org/documentation/info/family_member_symbol</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.features_id">
<code class="descname">features_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.features_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/features/:id</span></code></p>
<p>Return the GA4GH record for a specific sequence feature given its identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/features_id">https://rest.ensembl.org/documentation/info/features_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.features_post">
<code class="descname">features_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.features_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/features/search</span></code></p>
<p>Return a list of sequence annotation features in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/features_post">https://rest.ensembl.org/documentation/info/features_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.fetch_all_epigenomes">
<code class="descname">fetch_all_epigenomes</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.fetch_all_epigenomes" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/epigenome</span></code></p>
<p>Returns information about all epigenomes available for the given species
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/fetch_all_epigenomes">https://rest.ensembl.org/documentation/info/fetch_all_epigenomes</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gacallSet">
<code class="descname">gacallSet</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gacallSet" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/callsets/search</span></code></p>
<p>Return a list of sets of genotype calls for specific samples in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gacallSet">https://rest.ensembl.org/documentation/info/gacallSet</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gacallset_id">
<code class="descname">gacallset_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gacallset_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/callsets/:id</span></code></p>
<p>Return the GA4GH record for a specific CallSet given its identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gacallset_id">https://rest.ensembl.org/documentation/info/gacallset_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gadataset">
<code class="descname">gadataset</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gadataset" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/datasets/search</span></code></p>
<p>Return a list of datasets in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gadataset">https://rest.ensembl.org/documentation/info/gadataset</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gadataset_id">
<code class="descname">gadataset_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gadataset_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/datasets/:id</span></code></p>
<p>Return the GA4GH record for a specific dataset given its identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gadataset_id">https://rest.ensembl.org/documentation/info/gadataset_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gafeatureset">
<code class="descname">gafeatureset</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gafeatureset" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/featuresets/search</span></code></p>
<p>Return a list of feature sets in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gafeatureset">https://rest.ensembl.org/documentation/info/gafeatureset</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gafeatureset_id">
<code class="descname">gafeatureset_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gafeatureset_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/featuresets/:id</span></code></p>
<p>Return the GA4GH record for a specific featureSet given its identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gafeatureset_id">https://rest.ensembl.org/documentation/info/gafeatureset_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gavariant_id">
<code class="descname">gavariant_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gavariant_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/variants/:id</span></code></p>
<p>Return the GA4GH record for a specific variant given its identifier.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gavariant_id">https://rest.ensembl.org/documentation/info/gavariant_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gavariantannotations">
<code class="descname">gavariantannotations</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gavariantannotations" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/variantannotations/search</span></code></p>
<p>Return variant annotation information in GA4GH format for a region on a reference sequence
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gavariantannotations">https://rest.ensembl.org/documentation/info/gavariantannotations</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gavariants">
<code class="descname">gavariants</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gavariants" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/variants/search</span></code></p>
<p>Return variant call information in GA4GH format for a region on a reference sequence
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gavariants">https://rest.ensembl.org/documentation/info/gavariants</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gavariantset">
<code class="descname">gavariantset</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gavariantset" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/variantsets/search</span></code></p>
<p>Return a list of variant sets in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gavariantset">https://rest.ensembl.org/documentation/info/gavariantset</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.gavariantset_id">
<code class="descname">gavariantset_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.gavariantset_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/variantsets/:id</span></code></p>
<p>Return the GA4GH record for a specific VariantSet given its identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/gavariantset_id">https://rest.ensembl.org/documentation/info/gavariantset_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.genetree">
<code class="descname">genetree</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.genetree" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">genetree/id/:id</span></code></p>
<p>Retrieves a gene tree for a gene tree stable identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/genetree">https://rest.ensembl.org/documentation/info/genetree</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.genetree_member_id">
<code class="descname">genetree_member_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.genetree_member_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">genetree/member/id/:id</span></code></p>
<p>Retrieves the gene tree that contains the gene / transcript / translation stable identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/genetree_member_id">https://rest.ensembl.org/documentation/info/genetree_member_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.genetree_member_symbol">
<code class="descname">genetree_member_symbol</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.genetree_member_symbol" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">genetree/member/symbol/:species/:symbol</span></code></p>
<p>Retrieves the gene tree that contains the gene identified by a symbol
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/genetree_member_symbol">https://rest.ensembl.org/documentation/info/genetree_member_symbol</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.genomic_alignment_region">
<code class="descname">genomic_alignment_region</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.genomic_alignment_region" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">alignment/region/:species/:region</span></code></p>
<p>Retrieves genomic alignments as separate blocks based on a region and species
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/genomic_alignment_region">https://rest.ensembl.org/documentation/info/genomic_alignment_region</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.homology_ensemblgene">
<code class="descname">homology_ensemblgene</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.homology_ensemblgene" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">homology/id/:id</span></code></p>
<p>Retrieves homology information (orthologs) by Ensembl gene id
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/homology_ensemblgene">https://rest.ensembl.org/documentation/info/homology_ensemblgene</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.homology_symbol">
<code class="descname">homology_symbol</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.homology_symbol" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">homology/symbol/:species/:symbol</span></code></p>
<p>Retrieves homology information (orthologs) by symbol
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/homology_symbol">https://rest.ensembl.org/documentation/info/homology_symbol</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ld_id_get">
<code class="descname">ld_id_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ld_id_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ld/:species/:id/:population_name</span></code></p>
<p>Computes and returns LD values between the given variant and all other variants in a window centered around the given variant. The window size is set to 500 kb.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ld_id_get">https://rest.ensembl.org/documentation/info/ld_id_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ld_pairwise_get">
<code class="descname">ld_pairwise_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ld_pairwise_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ld/:species/pairwise/:id1/:id2</span></code></p>
<p>Computes and returns LD values between the given variants.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ld_pairwise_get">https://rest.ensembl.org/documentation/info/ld_pairwise_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ld_region_get">
<code class="descname">ld_region_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ld_region_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ld/:species/region/:region/:population_name</span></code></p>
<p>Computes and returns LD values between all pairs of variants in the defined region.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ld_region_get">https://rest.ensembl.org/documentation/info/ld_region_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.list_all_microarrays">
<code class="descname">list_all_microarrays</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.list_all_microarrays" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/microarray</span></code></p>
<p>Returns information about all microarrays available for the given species
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/list_all_microarrays">https://rest.ensembl.org/documentation/info/list_all_microarrays</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.lookup">
<code class="descname">lookup</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.lookup" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">lookup/id/:id</span></code></p>
<p>Find the species and database for a single identifier e.g. gene, transcript, protein
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/lookup">https://rest.ensembl.org/documentation/info/lookup</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.lookup_post">
<code class="descname">lookup_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.lookup_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">lookup/id</span></code></p>
<p>Find the species and database for several identifiers. IDs that are not found are returned with no data.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/lookup_post">https://rest.ensembl.org/documentation/info/lookup_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ontology_ancestors">
<code class="descname">ontology_ancestors</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ontology_ancestors" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/ancestors/:id</span></code></p>
<p>Reconstruct the entire ancestry of a term from is_a and part_of relationships
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ontology_ancestors">https://rest.ensembl.org/documentation/info/ontology_ancestors</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ontology_ancestors_chart">
<code class="descname">ontology_ancestors_chart</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ontology_ancestors_chart" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/ancestors/chart/:id</span></code></p>
<p>Reconstruct the entire ancestry of a term from is_a and part_of relationships.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ontology_ancestors_chart">https://rest.ensembl.org/documentation/info/ontology_ancestors_chart</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ontology_descendants">
<code class="descname">ontology_descendants</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ontology_descendants" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/descendants/:id</span></code></p>
<p>Find all the terms descended from a given term. By default searches are conducted within the namespace of the given identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ontology_descendants">https://rest.ensembl.org/documentation/info/ontology_descendants</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ontology_id">
<code class="descname">ontology_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ontology_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/id/:id</span></code></p>
<p>Search for an ontological term by its namespaced identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ontology_id">https://rest.ensembl.org/documentation/info/ontology_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ontology_name">
<code class="descname">ontology_name</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ontology_name" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/name/:name</span></code></p>
<p>Search for a list of ontological terms by their name
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ontology_name">https://rest.ensembl.org/documentation/info/ontology_name</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.overlap_id">
<code class="descname">overlap_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.overlap_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">overlap/id/:id</span></code></p>
<p>Retrieves features (e.g. genes, transcripts, variants and more) that overlap a region defined by the given identifier.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/overlap_id">https://rest.ensembl.org/documentation/info/overlap_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.overlap_region">
<code class="descname">overlap_region</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.overlap_region" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">overlap/region/:species/:region</span></code></p>
<p>Retrieves features (e.g. genes, transcripts, variants and more) that overlap a given region.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/overlap_region">https://rest.ensembl.org/documentation/info/overlap_region</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.overlap_translation">
<code class="descname">overlap_translation</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.overlap_translation" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">overlap/translation/:id</span></code></p>
<p>Retrieve features related to a specific Translation as described by its stable ID (e.g. domains, variants).
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/overlap_translation">https://rest.ensembl.org/documentation/info/overlap_translation</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.phenotype_accession">
<code class="descname">phenotype_accession</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.phenotype_accession" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">/phenotype/accession/:species/:accession</span></code></p>
<p>Return phenotype annotations for genomic features given a phenotype ontology accession
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/phenotype_accession">https://rest.ensembl.org/documentation/info/phenotype_accession</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.phenotype_gene">
<code class="descname">phenotype_gene</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.phenotype_gene" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">/phenotype/gene/:species/:gene</span></code></p>
<p>Return phenotype annotations for a given gene.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/phenotype_gene">https://rest.ensembl.org/documentation/info/phenotype_gene</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.phenotype_region">
<code class="descname">phenotype_region</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.phenotype_region" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">/phenotype/region/:species/:region</span></code></p>
<p>Return phenotype annotations that overlap a given genomic region.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/phenotype_region">https://rest.ensembl.org/documentation/info/phenotype_region</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.phenotype_term">
<code class="descname">phenotype_term</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.phenotype_term" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">/phenotype/term/:species/:term</span></code></p>
<p>Return phenotype annotations for genomic features given a phenotype ontology term
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/phenotype_term">https://rest.ensembl.org/documentation/info/phenotype_term</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.ping">
<code class="descname">ping</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.ping" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/ping</span></code></p>
<p>Checks if the service is alive.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/ping">https://rest.ensembl.org/documentation/info/ping</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.probe">
<code class="descname">probe</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.probe" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/microarray/:microarray/probe/:probe</span></code></p>
<p>Returns information about a specific probe from a microarray
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/probe">https://rest.ensembl.org/documentation/info/probe</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.probe_set">
<code class="descname">probe_set</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.probe_set" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/microarray/:microarray/probe_set/:probe_set</span></code></p>
<p>Returns information about a specific probe_set from a microarray
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/probe_set">https://rest.ensembl.org/documentation/info/probe_set</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.referenceSets">
<code class="descname">referenceSets</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.referenceSets" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/referencesets/search</span></code></p>
<p>Return a list of reference sets in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/referenceSets">https://rest.ensembl.org/documentation/info/referenceSets</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.referenceSets_id">
<code class="descname">referenceSets_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.referenceSets_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/referencesets/:id</span></code></p>
<p>Return data for a specific reference set in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/referenceSets_id">https://rest.ensembl.org/documentation/info/referenceSets_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.references">
<code class="descname">references</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.references" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">ga4gh/references/search</span></code></p>
<p>Return a list of reference sequences in GA4GH format
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/references">https://rest.ensembl.org/documentation/info/references</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.references_id">
<code class="descname">references_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.references_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ga4gh/references/:id</span></code></p>
<p>Return data for a specific reference in GA4GH format by id
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/references_id">https://rest.ensembl.org/documentation/info/references_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.regulatory_id">
<code class="descname">regulatory_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.regulatory_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/id/:id</span></code></p>
<p>Returns a RegulatoryFeature given its stable ID (e.g. ENSR00000099113)
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/regulatory_id">https://rest.ensembl.org/documentation/info/regulatory_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.rest">
<code class="descname">rest</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.rest" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/rest</span></code></p>
<p>Shows the current version of the Ensembl REST API.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/rest">https://rest.ensembl.org/documentation/info/rest</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.sequence_id">
<code class="descname">sequence_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.sequence_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">sequence/id/:id</span></code></p>
<p>Request multiple types of sequence by stable identifier. Supports feature masking and expand options.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/sequence_id">https://rest.ensembl.org/documentation/info/sequence_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.sequence_id_post">
<code class="descname">sequence_id_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.sequence_id_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">sequence/id</span></code></p>
<p>Request multiple types of sequence by a stable identifier list.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/sequence_id_post">https://rest.ensembl.org/documentation/info/sequence_id_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.sequence_region">
<code class="descname">sequence_region</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.sequence_region" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">sequence/region/:species/:region</span></code></p>
<p>Returns the genomic sequence of the specified region of the given species. Supports feature masking and expand options.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/sequence_region">https://rest.ensembl.org/documentation/info/sequence_region</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.sequence_region_post">
<code class="descname">sequence_region_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.sequence_region_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">sequence/region/:species</span></code></p>
<p>Request multiple types of sequence by a list of regions.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/sequence_region_post">https://rest.ensembl.org/documentation/info/sequence_region_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.software">
<code class="descname">software</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.software" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/software</span></code></p>
<p>Shows the current version of the Ensembl API used by the REST server.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/software">https://rest.ensembl.org/documentation/info/software</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.species">
<code class="descname">species</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.species" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/species</span></code></p>
<p>Lists all available species, their aliases, available adaptor groups and data release.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/species">https://rest.ensembl.org/documentation/info/species</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.species_id">
<code class="descname">species_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.species_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">eqtl/stable_id/:species/:stable_id</span></code></p>
<p>Returns the p-value for each SNP in a given gene (e.g. ENSG00000227232)
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/species_id">https://rest.ensembl.org/documentation/info/species_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.species_variant">
<code class="descname">species_variant</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.species_variant" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">eqtl/variant_name/:species/:variant_name</span></code></p>
<p>Returns the p-values for a SNP (e.g. rs123)
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/species_variant">https://rest.ensembl.org/documentation/info/species_variant</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.symbol_lookup">
<code class="descname">symbol_lookup</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.symbol_lookup" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">lookup/symbol/:species/:symbol</span></code></p>
<p>Find the species and database for a symbol in a linked external database
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/symbol_lookup">https://rest.ensembl.org/documentation/info/symbol_lookup</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.symbol_post">
<code class="descname">symbol_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.symbol_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">lookup/symbol/:species/:symbol</span></code></p>
<p>Find the species and database for a set of symbols in a linked external database. Unknown symbols are omitted from the response.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/symbol_post">https://rest.ensembl.org/documentation/info/symbol_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.taxonomy_classification">
<code class="descname">taxonomy_classification</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.taxonomy_classification" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">taxonomy/classification/:id</span></code></p>
<p>Return the taxonomic classification of a taxon node
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/taxonomy_classification">https://rest.ensembl.org/documentation/info/taxonomy_classification</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.taxonomy_id">
<code class="descname">taxonomy_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.taxonomy_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">taxonomy/id/:id</span></code></p>
<p>Search for a taxonomic term by its identifier or name
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/taxonomy_id">https://rest.ensembl.org/documentation/info/taxonomy_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.taxonomy_name">
<code class="descname">taxonomy_name</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.taxonomy_name" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">taxonomy/name/:name</span></code></p>
<p>Search for a taxonomic id by a non-scientific name
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/taxonomy_name">https://rest.ensembl.org/documentation/info/taxonomy_name</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.tissues">
<code class="descname">tissues</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.tissues" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">eqtl/tissue/:species/</span></code></p>
<p>Returns all tissues currently available in the DB
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/tissues">https://rest.ensembl.org/documentation/info/tissues</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.transcript_haplotypes_get">
<code class="descname">transcript_haplotypes_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.transcript_haplotypes_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">transcript_haplotypes/:species/:id</span></code></p>
<p>Computes observed transcript haplotype sequences based on phased genotype data
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/transcript_haplotypes_get">https://rest.ensembl.org/documentation/info/transcript_haplotypes_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variant_recoder">
<code class="descname">variant_recoder</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variant_recoder" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">variant_recoder/:species/:id</span></code></p>
<p>Translate a variant identifier or HGVS notation to all possible variant IDs and HGVS
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variant_recoder">https://rest.ensembl.org/documentation/info/variant_recoder</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variant_recoder_post">
<code class="descname">variant_recoder_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variant_recoder_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">variant_recoder/:species</span></code></p>
<p>Translate a list of variant identifiers or HGVS notations to all possible variant IDs and HGVS
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variant_recoder_post">https://rest.ensembl.org/documentation/info/variant_recoder_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variation">
<code class="descname">variation</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variation" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/variation/:species</span></code></p>
<p>List the variation sources used in Ensembl for a species.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variation">https://rest.ensembl.org/documentation/info/variation</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variation_consequence_types">
<code class="descname">variation_consequence_types</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variation_consequence_types" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/variation/consequence_types</span></code></p>
<p>Lists all variant consequence types.
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variation_consequence_types">https://rest.ensembl.org/documentation/info/variation_consequence_types</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variation_id">
<code class="descname">variation_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variation_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">variation/:species/:id</span></code></p>
<p>Uses a variant identifier (e.g. rsID) to return the variation features including optional genotype, phenotype and population data
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variation_id">https://rest.ensembl.org/documentation/info/variation_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variation_pmcid_get">
<code class="descname">variation_pmcid_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variation_pmcid_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">variation/:species/pmcid/:pmcid</span></code></p>
<p>Fetch variants by publication using PubMed Central reference number (PMCID)
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variation_pmcid_get">https://rest.ensembl.org/documentation/info/variation_pmcid_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variation_pmid_get">
<code class="descname">variation_pmid_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variation_pmid_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">variation/:species/pmid/:pmid</span></code></p>
<p>Fetch variants by publication using PubMed reference number (PMID)
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variation_pmid_get">https://rest.ensembl.org/documentation/info/variation_pmid_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variation_populations">
<code class="descname">variation_populations</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variation_populations" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/variation/populations/:species</span></code></p>
<p>List all populations for a species
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variation_populations">https://rest.ensembl.org/documentation/info/variation_populations</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.variation_post">
<code class="descname">variation_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.variation_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">variation/:species/</span></code></p>
<p>Uses a list of variant identifiers (e.g. rsID) to return the variation features including optional genotype, phenotype and population data
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/variation_post">https://rest.ensembl.org/documentation/info/variation_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.vep_hgvs_get">
<code class="descname">vep_hgvs_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.vep_hgvs_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">vep/:species/hgvs/:hgvs_notation</span></code></p>
<p>Fetch variant consequences based on a HGVS notation
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/vep_hgvs_get">https://rest.ensembl.org/documentation/info/vep_hgvs_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.vep_hgvs_post">
<code class="descname">vep_hgvs_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.vep_hgvs_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">vep/:species/hgvs</span></code></p>
<p>Fetch variant consequences for multiple HGVS notations
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/vep_hgvs_post">https://rest.ensembl.org/documentation/info/vep_hgvs_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.vep_id_get">
<code class="descname">vep_id_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.vep_id_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">vep/:species/id/:id</span></code></p>
<p>Fetch variant consequences based on a variant identifier
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/vep_id_get">https://rest.ensembl.org/documentation/info/vep_id_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.vep_id_post">
<code class="descname">vep_id_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.vep_id_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">vep/:species/id</span></code></p>
<p>Fetch variant consequences for multiple ids
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/vep_id_post">https://rest.ensembl.org/documentation/info/vep_id_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.vep_region_get">
<code class="descname">vep_region_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.vep_region_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">vep/:species/region/:region/:allele/</span></code></p>
<p>Fetch variant consequences
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/vep_region_get">https://rest.ensembl.org/documentation/info/vep_region_get</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.vep_region_post">
<code class="descname">vep_region_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.vep_region_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">vep/:species/region</span></code></p>
<p>Fetch variant consequences for multiple regions
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/vep_region_post">https://rest.ensembl.org/documentation/info/vep_region_post</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.xref_external">
<code class="descname">xref_external</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.xref_external" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">xrefs/symbol/:species/:symbol</span></code></p>
<p>Looks up an external symbol and returns all Ensembl objects linked to it. This can be a display name for a gene/transcript/translation, a synonym or an externally linked reference. If a gene’s transcript is linked to the supplied symbol the service will return both gene and transcript (it supports transient links).
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/xref_external">https://rest.ensembl.org/documentation/info/xref_external</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.xref_id">
<code class="descname">xref_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.xref_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">xrefs/id/:id</span></code></p>
<p>Perform lookups of Ensembl Identifiers and retrieve their external references in other databases
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/xref_id">https://rest.ensembl.org/documentation/info/xref_id</a></p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblClient.xref_name">
<code class="descname">xref_name</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblClient.xref_name" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">xrefs/name/:species/:name</span></code></p>
<p>Performs a lookup based upon the primary accession or display label of an external reference and returning the information we hold about the entry
- More info: <a class="reference external" href="https://rest.ensembl.org/documentation/info/xref_name">https://rest.ensembl.org/documentation/info/xref_name</a></p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="ensembl_rest.EnsemblGenomesClient">
<em class="property">class </em><code class="descclassname">ensembl_rest.</code><code class="descname">EnsemblGenomesClient</code><span class="sig-paren">(</span><em>base_url=None</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient" title="Permalink to this definition">¶</a></dt>
<dd><p>A client for the EnsemblGenomes REST API (<a class="reference external" href="http://rest.ensemblgenomes.org">http://rest.ensemblgenomes.org</a>)</p>
<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.analysis">
<code class="descname">analysis</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.analysis" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/analysis/:species</span></code></p>
<p>List the names of analyses involved in generating Ensembl data.
- More info: /documentation/info/analysis</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.array_info">
<code class="descname">array_info</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.array_info" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/microarray/:microarray/vendor/:vendor</span></code></p>
<p>Returns information about a specific microarray
- More info: /documentation/info/array_info</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.assembly_cdna">
<code class="descname">assembly_cdna</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.assembly_cdna" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">map/cdna/:id/:region</span></code></p>
<p>Convert from cDNA coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
- More info: /documentation/info/assembly_cdna</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.assembly_cds">
<code class="descname">assembly_cds</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.assembly_cds" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">map/cds/:id/:region</span></code></p>
<p>Convert from CDS coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
- More info: /documentation/info/assembly_cds</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.assembly_info">
<code class="descname">assembly_info</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.assembly_info" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/assembly/:species</span></code></p>
<p>List the currently available assemblies for a species, along with toplevel sequences, chromosomes and cytogenetic bands.
- More info: /documentation/info/assembly_info</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.assembly_map">
<code class="descname">assembly_map</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.assembly_map" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">map/:species/:asm_one/:region/:asm_two</span></code></p>
<p>Convert the co-ordinates of one assembly to another
- More info: /documentation/info/assembly_map</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.assembly_stats">
<code class="descname">assembly_stats</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.assembly_stats" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/assembly/:species/:region_name</span></code></p>
<p>Returns information about the specified toplevel sequence region for the given species.
- More info: /documentation/info/assembly_stats</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.assembly_translation">
<code class="descname">assembly_translation</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.assembly_translation" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">map/translation/:id/:region</span></code></p>
<p>Convert from protein (translation) coordinates to genomic coordinates. Output reflects forward orientation coordinates as returned from the Ensembl API.
- More info: /documentation/info/assembly_translation</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.biotypes">
<code class="descname">biotypes</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.biotypes" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/biotypes/:species</span></code></p>
<p>List the functional classifications of gene models that Ensembl associates with a particular species. Useful for restricting the type of genes/transcripts retrieved by other endpoints.
- More info: /documentation/info/biotypes</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.biotypes_groups">
<code class="descname">biotypes_groups</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.biotypes_groups" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/biotypes/groups/:group/:object_type</span></code></p>
<p>Without argument the list of available biotype groups is returned. With :group argument provided, list the properties of biotypes within that group. Object type (gene or transcript) can be provided for filtering.
- More info: /documentation/info/biotypes_groups</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.biotypes_name">
<code class="descname">biotypes_name</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.biotypes_name" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/biotypes/name/:name/:object_type</span></code></p>
<p>List the properties of biotypes with a given name. Object type (gene or transcript) can be provided for filtering.
- More info: /documentation/info/biotypes_name</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.compara_methods">
<code class="descname">compara_methods</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.compara_methods" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/compara/methods</span></code></p>
<p>List all compara analyses available (an analysis defines the type of comparative data).
- More info: /documentation/info/compara_methods</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.compara_species_sets">
<code class="descname">compara_species_sets</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.compara_species_sets" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/compara/species_sets/:method</span></code></p>
<p>List all collections of species analysed with the specified compara method.
- More info: /documentation/info/compara_species_sets</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.comparas">
<code class="descname">comparas</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.comparas" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/comparas</span></code></p>
<p>Lists all available comparative genomics databases and their data release.
- More info: /documentation/info/comparas</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.data">
<code class="descname">data</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.data" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/data</span></code></p>
<p>Shows the data releases available on this REST server. May return more than one release (unfrequent non-standard Ensembl configuration).
- More info: /documentation/info/data</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.eg_version">
<code class="descname">eg_version</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.eg_version" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/eg_version</span></code></p>
<p>Returns the Ensembl Genomes version of the databases backing this service
- More info: /documentation/info/eg_version</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.external_dbs">
<code class="descname">external_dbs</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.external_dbs" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/external_dbs/:species</span></code></p>
<p>Lists all available external sources for a species.
- More info: /documentation/info/external_dbs</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.family">
<code class="descname">family</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.family" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">family/id/:id</span></code></p>
<p>Retrieves gene family information by ID
- More info: /documentation/info/family</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.family_member_id">
<code class="descname">family_member_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.family_member_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">family/member/id/:id</span></code></p>
<p>Retrieves gene families to which a gene belongs
- More info: /documentation/info/family_member_id</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.family_member_symbol">
<code class="descname">family_member_symbol</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.family_member_symbol" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">family/member/symbol/:species/:symbol</span></code></p>
<p>Retrieves gene families to which a gene belongs
- More info: /documentation/info/family_member_symbol</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.genetree">
<code class="descname">genetree</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.genetree" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">genetree/id/:id</span></code></p>
<p>Retrieves a gene tree dump for a gene tree stable identifier
- More info: /documentation/info/genetree</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.genetree_member_id">
<code class="descname">genetree_member_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.genetree_member_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">genetree/member/id/:id</span></code></p>
<p>Retrieves a gene tree that contains the stable identifier
- More info: /documentation/info/genetree_member_id</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.genetree_member_symbol">
<code class="descname">genetree_member_symbol</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.genetree_member_symbol" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">genetree/member/symbol/:species/:symbol</span></code></p>
<p>Retrieves a gene tree containing the gene identified by a symbol
- More info: /documentation/info/genetree_member_symbol</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.genomic_alignment_region">
<code class="descname">genomic_alignment_region</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.genomic_alignment_region" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">alignment/region/:species/:region</span></code></p>
<p>Retrieves genomic alignments as separate blocks based on a region and species
- More info: /documentation/info/genomic_alignment_region</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.homology_ensemblgene">
<code class="descname">homology_ensemblgene</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.homology_ensemblgene" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">homology/id/:id</span></code></p>
<p>Retrieves homology information (orthologs) by Ensembl gene id
- More info: /documentation/info/homology_ensemblgene</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.homology_symbol">
<code class="descname">homology_symbol</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.homology_symbol" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">homology/symbol/:species/:symbol</span></code></p>
<p>Retrieves homology information (orthologs) by symbol
- More info: /documentation/info/homology_symbol</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.info_divisions">
<code class="descname">info_divisions</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.info_divisions" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/divisions</span></code></p>
<p>Get list of all Ensembl divisions for which information is available
- More info: /documentation/info/info_divisions</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.info_genome">
<code class="descname">info_genome</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.info_genome" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/genomes/:genome_name</span></code></p>
<p>Find information about a given genome
- More info: /documentation/info/info_genome</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.info_genomes">
<code class="descname">info_genomes</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.info_genomes" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/genomes</span></code></p>
<p>Find information about all genomes. Response may be very large.
- More info: /documentation/info/info_genomes</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.info_genomes_accession">
<code class="descname">info_genomes_accession</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.info_genomes_accession" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/genomes/accession/:accession</span></code></p>
<p>Find information about genomes containing a specified INSDC accession
- More info: /documentation/info/info_genomes_accession</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.info_genomes_assembly">
<code class="descname">info_genomes_assembly</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.info_genomes_assembly" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/genomes/assembly/:assembly_id</span></code></p>
<p>Find information about a genome with a specified assembly
- More info: /documentation/info/info_genomes_assembly</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.info_genomes_division">
<code class="descname">info_genomes_division</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.info_genomes_division" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/genomes/division/:division_name</span></code></p>
<p>Find information about all genomes in a given division. May be large for Ensembl Bacteria.
- More info: /documentation/info/info_genomes_division</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.info_genomes_taxonomy">
<code class="descname">info_genomes_taxonomy</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.info_genomes_taxonomy" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/genomes/taxonomy/:taxon_name</span></code></p>
<p>Find information about all genomes beneath a given node of the taxonomy
- More info: /documentation/info/info_genomes_taxonomy</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.list_all_microarrays">
<code class="descname">list_all_microarrays</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.list_all_microarrays" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/microarray</span></code></p>
<p>Returns information about all microarrays available for the given species
- More info: /documentation/info/list_all_microarrays</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.lookup">
<code class="descname">lookup</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.lookup" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">lookup/id/:id</span></code></p>
<p>Find the species and database for a single identifier e.g. gene, transcript, protein
- More info: /documentation/info/lookup</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.lookup_genome">
<code class="descname">lookup_genome</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.lookup_genome" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">lookup/genome/:name</span></code></p>
<p>Query for a named genome and retrieve the gene models
- More info: /documentation/info/lookup_genome</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.lookup_post">
<code class="descname">lookup_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.lookup_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">lookup/id</span></code></p>
<p>Find the species and database for several identifiers. IDs that are not found are returned with no data.
- More info: /documentation/info/lookup_post</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.ontology_ancestors">
<code class="descname">ontology_ancestors</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.ontology_ancestors" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/ancestors/:id</span></code></p>
<p>Reconstruct the entire ancestry of a term from is_a and part_of relationships
- More info: /documentation/info/ontology_ancestors</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.ontology_ancestors_chart">
<code class="descname">ontology_ancestors_chart</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.ontology_ancestors_chart" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/ancestors/chart/:id</span></code></p>
<p>Reconstruct the entire ancestry of a term from is_a and part_of relationships.
- More info: /documentation/info/ontology_ancestors_chart</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.ontology_descendants">
<code class="descname">ontology_descendants</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.ontology_descendants" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/descendants/:id</span></code></p>
<p>Find all the terms descended from a given term. By default searches are conducted within the namespace of the given identifier
- More info: /documentation/info/ontology_descendants</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.ontology_id">
<code class="descname">ontology_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.ontology_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/id/:id</span></code></p>
<p>Search for an ontological term by its namespaced identifier
- More info: /documentation/info/ontology_id</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.ontology_name">
<code class="descname">ontology_name</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.ontology_name" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">ontology/name/:name</span></code></p>
<p>Search for a list of ontological terms by their name
- More info: /documentation/info/ontology_name</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.overlap_id">
<code class="descname">overlap_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.overlap_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">overlap/id/:id</span></code></p>
<p>Retrieves features (e.g. genes, transcripts, variants and more) that overlap a region defined by the given identifier.
- More info: /documentation/info/overlap_id</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.overlap_region">
<code class="descname">overlap_region</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.overlap_region" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">overlap/region/:species/:region</span></code></p>
<p>Retrieves features (e.g. genes, transcripts, variants and more) that overlap a given region.
- More info: /documentation/info/overlap_region</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.overlap_translation">
<code class="descname">overlap_translation</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.overlap_translation" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">overlap/translation/:id</span></code></p>
<p>Retrieve features related to a specific Translation as described by its stable ID (e.g. domains, variants).
- More info: /documentation/info/overlap_translation</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.phenotype_gene">
<code class="descname">phenotype_gene</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.phenotype_gene" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">/phenotype/gene/:species/:gene</span></code></p>
<p>Return phenotype annotations for a given gene.
- More info: /documentation/info/phenotype_gene</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.phenotype_region">
<code class="descname">phenotype_region</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.phenotype_region" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">/phenotype/region/:species/:region</span></code></p>
<p>Return phenotype annotations that overlap a given genomic region.
- More info: /documentation/info/phenotype_region</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.ping">
<code class="descname">ping</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.ping" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/ping</span></code></p>
<p>Checks if the service is alive.
- More info: /documentation/info/ping</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.probe_info">
<code class="descname">probe_info</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.probe_info" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">regulatory/species/:species/microarray/:microarray/probe/:probe</span></code></p>
<p>Returns information about a specific probe of a microarray
- More info: /documentation/info/probe_info</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.rest">
<code class="descname">rest</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.rest" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/rest</span></code></p>
<p>Shows the current version of the Ensembl REST API.
- More info: /documentation/info/rest</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.sequence_id">
<code class="descname">sequence_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.sequence_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">sequence/id/:id</span></code></p>
<p>Request multiple types of sequence by stable identifier. Supports feature masking and expand options.
- More info: /documentation/info/sequence_id</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.sequence_id_post">
<code class="descname">sequence_id_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.sequence_id_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">sequence/id</span></code></p>
<p>Request multiple types of sequence by a stable identifier list.
- More info: /documentation/info/sequence_id_post</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.sequence_region">
<code class="descname">sequence_region</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.sequence_region" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">sequence/region/:species/:region</span></code></p>
<p>Returns the genomic sequence of the specified region of the given species. Supports feature masking and expand options.
- More info: /documentation/info/sequence_region</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.sequence_region_post">
<code class="descname">sequence_region_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.sequence_region_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">sequence/region/:species</span></code></p>
<p>Request multiple types of sequence by a list of regions.
- More info: /documentation/info/sequence_region_post</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.software">
<code class="descname">software</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.software" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/software</span></code></p>
<p>Shows the current version of the Ensembl API used by the REST server.
- More info: /documentation/info/software</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.species">
<code class="descname">species</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.species" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/species</span></code></p>
<p>Lists all available species, their aliases, available adaptor groups and data release.
- More info: /documentation/info/species</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.symbol_lookup">
<code class="descname">symbol_lookup</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.symbol_lookup" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">lookup/symbol/:species/:symbol</span></code></p>
<p>Find the species and database for a symbol in a linked external database
- More info: /documentation/info/symbol_lookup</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.symbol_post">
<code class="descname">symbol_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.symbol_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">lookup/symbol/:species/:symbol</span></code></p>
<p>Find the species and database for a set of symbols in a linked external database. Unknown symbols are omitted from the response.
- More info: /documentation/info/symbol_post</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.taxonomy_classification">
<code class="descname">taxonomy_classification</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.taxonomy_classification" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">taxonomy/classification/:id</span></code></p>
<p>Return the taxonomic classification of a taxon node
- More info: /documentation/info/taxonomy_classification</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.taxonomy_id">
<code class="descname">taxonomy_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.taxonomy_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">taxonomy/id/:id</span></code></p>
<p>Search for a taxonomic term by its identifier or name
- More info: /documentation/info/taxonomy_id</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.taxonomy_name">
<code class="descname">taxonomy_name</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.taxonomy_name" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">taxonomy/name/:name</span></code></p>
<p>Search for a taxonomic id by a non-scientific name
- More info: /documentation/info/taxonomy_name</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.variant_recoder">
<code class="descname">variant_recoder</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.variant_recoder" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">variant_recoder/:species/:id</span></code></p>
<p>Translate a variant identifier or HGVS notation to all possible variant IDs and HGVS
- More info: /documentation/info/variant_recoder</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.variant_recoder_post">
<code class="descname">variant_recoder_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.variant_recoder_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">variant_recoder/:species</span></code></p>
<p>Translate a list of variant identifiers or HGVS notations to all possible variant IDs and HGVS
- More info: /documentation/info/variant_recoder_post</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.variation">
<code class="descname">variation</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.variation" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/variation/:species</span></code></p>
<p>List the variation sources used in Ensembl for a species.
- More info: /documentation/info/variation</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.variation_consequence_types">
<code class="descname">variation_consequence_types</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.variation_consequence_types" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/variation/consequence_types</span></code></p>
<p>Lists all variant consequence types.
- More info: /documentation/info/variation_consequence_types</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.variation_id">
<code class="descname">variation_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.variation_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">variation/:species/:id</span></code></p>
<p>Uses a variant identifier (e.g. rsID) to return the variation features including optional genotype, phenotype and population data
- More info: /documentation/info/variation_id</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.variation_populations">
<code class="descname">variation_populations</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.variation_populations" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">info/variation/populations/:species</span></code></p>
<p>List all populations for a species
- More info: /documentation/info/variation_populations</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.variation_post">
<code class="descname">variation_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.variation_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">variation/:species/</span></code></p>
<p>Uses a list of variant identifiers (e.g. rsID) to return the variation features including optional genotype, phenotype and population data
- More info: /documentation/info/variation_post</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.vep_id_get">
<code class="descname">vep_id_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.vep_id_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">vep/:species/id/:id</span></code></p>
<p>Fetch variant consequences based on a variant identifier
- More info: /documentation/info/vep_id_get</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.vep_id_post">
<code class="descname">vep_id_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.vep_id_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">vep/:species/id</span></code></p>
<p>Fetch variant consequences for multiple ids
- More info: /documentation/info/vep_id_post</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.vep_region_get">
<code class="descname">vep_region_get</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.vep_region_get" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">vep/:species/region/:region/:allele/</span></code></p>
<p>Fetch variant consequences
- More info: /documentation/info/vep_region_get</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.vep_region_post">
<code class="descname">vep_region_post</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.vep_region_post" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">POST</span> <span class="pre">vep/:species/region</span></code></p>
<p>Fetch variant consequences for multiple regions
- More info: /documentation/info/vep_region_post</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.xref_external">
<code class="descname">xref_external</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.xref_external" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">xrefs/symbol/:species/:symbol</span></code></p>
<p>Looks up an external symbol and returns all Ensembl objects linked to it. This can be a display name for a gene/transcript/translation, a synonym or an externally linked reference. If a gene’s transcript is linked to the supplied symbol the service will return both gene and transcript (it supports transient links).
- More info: /documentation/info/xref_external</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.xref_id">
<code class="descname">xref_id</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.xref_id" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">xrefs/id/:id</span></code></p>
<p>Perform lookups of Ensembl Identifiers and retrieve their external references in other databases
- More info: /documentation/info/xref_id</p>
</dd></dl>

<dl class="method">
<dt id="ensembl_rest.EnsemblGenomesClient.xref_name">
<code class="descname">xref_name</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#ensembl_rest.EnsemblGenomesClient.xref_name" title="Permalink to this definition">¶</a></dt>
<dd><p><code class="docutils literal"><span class="pre">GET</span> <span class="pre">xrefs/name/:species/:name</span></code></p>
<p>Performs a lookup based upon the primary accession or display label of an external reference and returning the information we hold about the entry
- More info: /documentation/info/xref_name</p>
</dd></dl>

</dd></dl>

</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper"><div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="#">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Andrés García (Ad115).
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.5</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>